internal/pkg/manifest/svc.go

```diff
@@ -73,18 +73,68 @@ func (a *Count) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 	}
 
-	if !a.Autoscaling.IsEmpty() {
+	if !c.Autoscaling.IsValid() {
+		return errUnmarshalSpot
+	}
+
+	if !c.Autoscaling.IsEmpty() {
+		// Successfully unmarshalled Autoscaling fields, return
 		return nil
 	}
 
-	if err := unmarshal(&a.Value); err != nil {
+	if err := unmarshal(&c.Value); err != nil {
 		return errUnmarshalCountOpts
 	}
 	return nil
 }
 
+// Spot is a custom type that can either be represented as a boolean, when
+// specified with a Range, or as an integer, which indicates the `desiredCount`
+// on a service that does not use application autoscaling.
+type Spot struct {
+	Enabled *bool
+	Base    *int
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the Spot
+// struct, allowing it to be unmarshalled into either an int value provided by
+// Base or a boolean value provided by Enabled.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (s *Spot) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&s.Base); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if *s.Base != 0 {
```

Comment:
I'm wondering if `s.Base != nil` would do? I remember `s.Base` would be `nil` if unmarshal fails, and we probably won't be able to dereference it as a result. Or is this needed because we don't want the users to put `0` for `spot`?

Reply:
It does not -- I tried :( And yeah it's because the default value for a nil pointer is the zero value, which is 0 in this case.

Reply:
Maybe this just to be safe (in case `yaml` changes it's behavior)
```suggestion
	if aws.StringValue(s.Base) != 0 {
```